aws_lambda_powertools/logging/types.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+import sys
+
+if sys.version_info < (3, 11):
+    from typing_extensions import NotRequired, TypedDict
+else:
+    from typing import NotRequired, TypedDict
```

Comment:
You'd need an additional check for 3.8 - when TypedDict was introduced, and removed from typing_extensions.